meilisearch/tests/search/locales.rs

```diff
@@ -0,0 +1,662 @@
+    ])
+});
+
+#[actix_rt::test]
+async fn simple_search() {
+    let server = Server::new().await;
+
+    let index = server.index("test");
+    let documents = DOCUMENTS.clone();
+    index
+        .update_settings(
+            json!({"searchableAttributes": ["name_en", "name_ja", "name_zh", "author_en", "author_ja", "author_zh", "description_en", "description_ja", "description_zh"]}),
+        )
+        .await;
+    index.add_documents(documents, None).await;
+    index.wait_task(1).await;
+
+    // english
+    index
+        .search(json!({"q": "Atta", "attributesToRetrieve": ["id"]}), |response, code| {
+            snapshot!(response, @r###"
+            {
+              "hits": [
+                {
+                  "id": 852
+                }
+              ],
+              "query": "Atta",
+              "processingTimeMs": "[duration]",
+              "limit": 20,
+              "offset": 0,
+              "estimatedTotalHits": 1
+            }
+            "###);
+            snapshot!(code, @"200 OK");
+        })
+        .await;
+
+    // japanese
+    index
+        .search(json!({"q": "進撃", "attributesToRetrieve": ["id"]}), |response, code| {
+            snapshot!(response, @r###"
+            {
+              "hits": [
+                {
+                  "id": 853
+                }
+              ],
+              "query": "進撃",
+              "processingTimeMs": "[duration]",
+              "limit": 20,
+              "offset": 0,
+              "estimatedTotalHits": 1
+            }
+            "###);
+            snapshot!(code, @"200 OK");
+        })
+        .await;
+
+    index
+        .search(
+            json!({"q": "進撃", "attributesToRetrieve": ["id"], "locales": ["jpn"]}),
+            |response, code| {
+                snapshot!(response, @r###"
+            {
+              "hits": [
+                {
+                  "id": 852
+                },
+                {
+                  "id": 853
+                }
+              ],
+              "query": "進撃",
+              "processingTimeMs": "[duration]",
+              "limit": 20,
+              "offset": 0,
+              "estimatedTotalHits": 2
+            }
+            "###);
+                snapshot!(code, @"200 OK");
+            },
+        )
+        .await;
+
+    // chinese
+    index
+        .search(json!({"q": "进击", "attributesToRetrieve": ["id"]}), |response, code| {
+            snapshot!(response, @r###"
+            {
+              "hits": [
+                {
+                  "id": 853
+                },
+                {
+                  "id": 852
+                }
+              ],
+              "query": "进击",
+              "processingTimeMs": "[duration]",
+              "limit": 20,
+              "offset": 0,
+              "estimatedTotalHits": 2
+            }
+            "###);
+            snapshot!(code, @"200 OK");
+        })
+        .await;
+}
+
+#[actix_rt::test]
+async fn force_locales() {
```

Comment:
I think it would be great to have an additional test that has two distinct rules

```
                "localizedAttributes": [
                    {"attributePatterns": ["*_ja"], "locales": ["jpn"]},
                    {"attributePatterns": ["*_zh"], "locales": ["cmn"]}
                ]